client/reader/discover/site-attribution.jsx

```diff
@@ -14,63 +16,75 @@ import { translate } from 'i18n-calypso';
 import FollowButton from 'reader/follow-button';
 import { getLinkProps } from './helper';
 import { recordFollowToggle, recordSiteClick } from './stats';
+import { getSiteUrl, getSourceFollowUrl, getSourceData } from 'reader/discover/helper';
+import SiteIcon from 'blocks/site-icon';
+import { getSite } from 'state/reader/sites/selectors';
+import QueryReaderSite from 'components/data/query-reader-site';
 
 class DiscoverSiteAttribution extends React.Component {
 	static propTypes = {
-		attribution: PropTypes.shape( {
-			blog_name: PropTypes.string.isRequired,
-			blog_url: PropTypes.string.isRequired,
-			avatar_url: PropTypes.string,
-		} ).isRequired,
-		siteUrl: PropTypes.string.isRequired,
-		followUrl: PropTypes.string.isRequired,
+		post: PropTypes.object.isRequired,
 	};
 
 	onSiteClick = () => recordSiteClick( this.props.siteUrl );
 
 	onFollowToggle = isFollowing => recordFollowToggle( isFollowing, this.props.siteUrl );
 
 	render() {
-		const attribution = this.props.attribution;
-		const classes = classNames( 'discover-attribution is-site', {
-			'is-missing-avatar': ! attribution.avatar_url,
-		} );
-		const siteLinkProps = getLinkProps( this.props.siteUrl );
+		const { post, site } = this.props;
+		const attribution = get( post, 'discover_metadata.attribution' );
+		const siteUrl = getSiteUrl( post );
+		const followUrl = getSourceFollowUrl( post );
+		const { blogId: siteId } = getSourceData( post );
+		const siteLinkProps = getLinkProps( siteUrl );
 		const siteClasses = classNames( 'discover-attribution__blog ignore-click' );
 
+		let avatarUrl = attribution.avatar_url;
+		// Drop default avatar
+		if ( path.basename( avatarUrl ) === 'defaultavatar.png' ) {
+			avatarUrl = null;
+		}
+
+		/* eslint-disable wpcalypso/jsx-classname-namespace */
 		return (
-			<div className={ classes }>
-				{ attribution.avatar_url ? (
+			<div className="discover__attribution is-site">
+				{ avatarUrl && (
 					<img
 						className="gravatar"
 						src={ encodeURI( attribution.avatar_url ) }
 						alt="Avatar"
 						width="20"
 						height="20"
 					/>
-				) : (
-					<Gridicon icon="globe" size={ 18 } />
 				) }
+				{ ! avatarUrl && siteId && <QueryReaderSite siteId={ siteId } /> }
```

Comment:
should we also hide the `QueryReaderSite` in the case that the site is already loaded?
i.e.
`! avatarUrl && siteId && ! site && ...`